pages/edit-team-details.js

```diff
@@ -0,0 +1,183 @@
+import fetch from '../lib/utils/api'
+import TeamDetailsForm from '../components/teams/TeamDetailsForm'
+import TeamMembersForm from '../components/teams/TeamMembersForm'
+import TeamCampaignsForm from '../components/teams/TeamCampaignsForm'
+import Router from '../lib/router'
+
+function DestroyButton ({ onDestroy }) {
+  const [destroyConfirmation, setDestroyConfirmation] = useState(false)
+
+  const destroyConfirmationButton = <div className='form__footer'>
+    <p>Are you sure you want to delete this team?</p>
+    <button className='button button--destroy'
+      id='delete-badge-confirmation-operation-button'
+      type='button'
+      onClick={onDestroy}
+    >
+      Delete this team permanently
+    </button>
+
+    <button className='button button--secondary'
+      id='cancel-delete-badge-operation-button'
+      type='button'
+      onClick={() => setDestroyConfirmation(false)} >
+      Cancel
+    </button>
+  </div>
+
+  const destroyButton = (
+    <button className='button button--destroy'
+      id='delete-badge-operation-button'
+      type='button'
+      onClick={() => setDestroyConfirmation(true)}
+    >
+      Delete this team
+    </button>
+  )
+
+  if (destroyConfirmation) return destroyConfirmationButton
+  else return destroyButton
+}
+
+export function EditTeam (props) {
+  /**
+   * Function for adding removing members from
+   * state
+   */
+  const [members, setMembers] = useReducer(
+    (members, { type, value }) => {
+      switch (type) {
+        case 'add':
+          return [...members, value]
+        case 'remove':
+          return members.filter(({ osm_id }) => (osm_id.toString() !== value.osm_id.toString()))
+      }
+    }, props.data.users || []
+  )
+
+  /**
+   * Function for adding removing campaigns from
+   * state
+   */
+  const [campaigns, setCampaigns] = useReducer(
+    (campaigns, { type, value }) => {
+      switch (type) {
+        case 'add': {
+          const uniqueCampaigns = campaigns.filter(
+            c => c.id !== value.id
+          )
+          return [...uniqueCampaigns, value]
+        }
+        case 'remove':
+          return campaigns.filter(({ id }) => id !== value.id)
+      }
+    }, props.data.campaigns || []
+  )
+
+  /**
+   * Function to update details
+   */
+  const [details, setDetails] = useState(
+    pick(['location', 'name', 'hashtag', 'bio'], props.data)
+  )
+
+  const handleSubmit = async () => {
+    await props.updateTeam(props.id, { ...details,
+      newusers: members.map(prop('osm_id')),
+      oldusers: props.data.users.map(prop('osm_id')),
+      campaigns: campaigns.map(pick(['id', 'team_priority']))
+    })
+    Router.push(`/teams/${props.id}`)
+  }
+
+  const handleDestroy = async () => {
+    await props.deleteTeam(props.id)
+    Router.push('/teams')
+  }
+
+  const [loading, setLoading] = useState(true)
+
+  // On load get the user
+  useEffect(() => {
+    props.getAuthenticatedUser()
+      .then(() => setLoading(false))
+  })
```

Comment:
```suggestion
  useEffect(() => {
    props.getAuthenticatedUser()
      .then(() => setLoading(false))
  }, [])
```

@kamicut consider using the ["useEffect once"](https://css-tricks.com/run-useeffect-only-once/) trick, because the form is pegging my cpu, and it's repeatedly hitting the `api/users/{id}` endpoint, I guess because of a state-update loop.

Checking for expired token or session etc, will have to be done application-wide in a consistent way. This change at least will warn the user if they are not signed in at the time they start editing in the form.